QgisModelBaker/utils/tools.py

```diff
@@ -0,0 +1,271 @@
+
+        if running:
+            busy_bar = QProgressBar()
+            busy_bar.setRange(0, 0)
+            busy_bar.setVisible(True)
+            layout.addWidget(busy_bar)
+
+        text_label = QLabel()
+        text_label.setText(message)
+        layout.addWidget(text_label)
+
+        if show_log_button:
+            button = QPushButton()
+            button.clicked.connect(lambda: self.parent.show_logs_folder())
+            button.setText(self.tr("Show log folder"))
+            layout.addWidget(button)
+
+        message_widget = QWidget()
+        message_widget.setLayout(layout)
+
+        self.message_bar.clearWidgets()
+        self.message_bar.pushWidget(message_widget, level, timeout)
+        self.log(f"Push to message bar: {message}", level)
+
+    def handle_dropped_files(self, dropped_files):
+
+        """
+        Main function receiving the files and trigger import to single GeoPackages.
+        After that it generates the layer via modelbaker generator.
+        """
+
+        data_files = []
+        model_files = []
+
+        # separate data files from ini files
+        for file in dropped_files:
+            if os.path.isfile(file):
+                if pathlib.Path(file).suffix[1:] in gui_utils.TransferExtensions:
+                    data_files.append(file)
+                else:
+                    model_files.append(file)
+
+        # one GeoPackage per file
+        status_map = {}
+        suc_files = set()
+        failed_files = set()
+
+        if len(data_files) == 0:
+            self.push_message_bar(
+                self.tr(
+                    "Nothing to import with Quick XTF Backer (use the Model Baker wizard instead)."
+                ),
+                False,
+                Qgis.MessageLevel.Warning,
+                15,
+            )
+            return suc_files, failed_files
+
+        if len(model_files):
+            self.log(
+                f"Dropped model files are ignored {model_files} (but if not available in repo, they are used because they are in the same directory as the tranfer files)."
+            )
+
+        self.push_message_bar(
+            self.tr("Quick'n'dirty XTF import with QuickVisualizer starts..."), True
+        )
+
+        for file in data_files:
+            self.push_message_bar(self.tr("Import {}").format(file), True)
+            status, db_file = self.import_file(file, self.parent.ili2db_configuration)
+            status_map[file] = {}
+            status_map[file]["status"] = status
+            status_map[file]["db_file"] = db_file
+            if status:
+                suc_files.add(file)
+            else:
+                failed_files.add(file)
+
+        # this is a fallback: in case we have a custom model directory defined, it could be that we didn't found the model in the tranferfile's directory
+        if (
+            len(failed_files) > 0
+            and self.parent.ili2db_configuration.custom_model_directories_enabled
+        ):
```

Comment:
Would it make sense for you to check if `XTF_DIR` is not yet in `custom_model_directories` before the fallback, so that perhaps we skip re-attempting if the `XTF_DIR` was already taken into account. That is, something like:

        if (
            len(failed_files) > 0
            and self.parent.ili2db_configuration.custom_model_directories_enabled
            and "%XTF_DIR" not in base_config.custom_model_directories
        ):

Reply:
good idea